sirepo/auth/email.py

```diff
@@ -67,10 +68,7 @@ def api_authEmailAuthorized(simulation_type, token):
         u = AuthEmailUser.search_by(token=token)
         if u and u.expires >= srtime.utc_now():
             n = _verify_confirm(req.type, token, auth.need_complete_registration(u))
-            u.query.filter(
-                (AuthEmailUser.user_name == u.unverified_email),
-                AuthEmailUser.unverified_email != u.unverified_email,
-            ).delete()
+            AuthEmailUser.delete_all_by_filter('changed_email', u)
```

Comment:
@robnagler this is kind of funky but I'm not sure how to improve it. Since we don't want to expose the _session outside of auth_db this the only way I could come up with performing the operation.

One possibe other option is to make _session a class property of UserDBBase. Then we could create a classmethod on AuthEmailIUser called delete_all_changed_email (or something to that effect) and use the session within it.
```
# auth_db.py
class UserDbBase:
    _session = _session
...

#auth/email.py
class AuthEmailUser:
...
    @classmethod
    def delete_all_changed_email(cls, user):
        with auth_db.thread_lock:
            _session.query(cls).filter_by(cls.user_name == user.unverified_email and cls.unverified_email != user.unverified_eamil).delete()
           _session.commit()

```

Reply:
I think this is clearer to use _session in the subclass.

I don't like `@sqlalchemy.ext.hybrid.hybrid_method`. It is even funkier than the usual `==` hack in sqlalchemy. It's completely unclear what's being returned from this method.

Reply:
Agreed, changed.